pandas/core/frame.py

```diff
@@ -2291,6 +2291,16 @@ def maybe_reorder(
             arr_columns = ensure_index(arr_columns)
             if columns is None:
                 columns = arr_columns
+
+                if len(arrays) == 0:
+                    if index is not None:
+                        if isinstance(index, str) or not hasattr(index, "__iter__"):
+                            if index not in columns:
+                                # error: Incompatible types in assignment
+                                # type "List[Any]"; expected "Optional[Index]"
+                                result_index = [index]  # type: ignore[assignment]
+                                index = None
+
```

Comment:
I would use only one `if` here to be consistent with other code.
```suggestion
                if (
                    len(arrays) == 0
                    and index is not None
                    and (isinstance(index, str) or not hasattr(index, "__iter__"))
                    and index not in columns
                ):
                    # error: Incompatible types in assignment
                    # type "List[Any]"; expected "Optional[Index]"
                    result_index = [index]  # type: ignore[assignment]
                    index = None
```

Reply:
I did check that if condition and earlier I had both conditions with one if but it was failing for other case where we pass this  pd.DataFrame.from_records([]) (it was failing for if index is not None condition)

Basically we can’t add both conditions in one if to filter out specific edge case.